src/pack.c

```diff
@@ -39,28 +39,31 @@
 
 #include "swupd.h"
 
-static void empty_pack_stage(int full, int version, char *module)
+static void empty_pack_stage(int full, int from_version, int to_version, char *module)
 {
 	char *cmd;
 	char *path;
 	int ret;
 
 	// clean any stale data (eg: re-run after a failure)
-	string_or_die(&cmd, "rm -rf %s/%s/%i/", packstage_dir, module, version);
+	string_or_die(&cmd, "rm -rf %s/%s/%i_to_%i/", packstage_dir, module,
+		      from_version, to_version);
 	ret = system(cmd);
 	if (ret) {
-		fprintf(stderr, "Failed to clean %s/%s/%i\n",
-			packstage_dir, module, version);
+		fprintf(stderr, "Failed to clean %s/%s/%i_to_%i\n",
+			packstage_dir, module, from_version, to_version);
 		exit(EXIT_FAILURE);
```

Comment:
@bryteise Thanks, I will add the free().

@ikeydoherty That would be a nice general cleanup, since string_or_die() is used all over the place. I'll take a look at your implementation.